### PR TITLE
Project SPA: Update to main navigation

### PR DESCRIPTION
**Pull request by joshsadam**
## Description of changes

Update to the main navigation to work on the SPA without any page variable.  Major update includes updated search input and users' initials as the dropdown to their account and logout (common pattern).  Removed the dropdown for announcements as this caused a lot of issues with each update for AntD.

This new navigation was also moved over into the rest of IRIDA to prevent different menus from being displayed.

## Related issue

N/A

## Checklist

Things for the developer to confirm they've done before the PR should be accepted:

*   [ ] ~CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.~
*   [ ] ~Tests added (or description of how to test) for any new features.~
*   [ ] ~User documentation updated for UI or technical changes.~

### CODE REVIEW

File: src/main/webapp/resources/js/components/main-navigation/components/CurrentUser.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  return (
+    <Avatar
+      style={{
+        backgroundColor: colour.background,
+        color: colour.text,
+        verticalAlign: "middle",
+      }}
+    >
+      {`${user?.firstName.charAt(0)}${user?.lastName.charAt(0)}`}
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/components/main-navigation/components/CurrentUser.tsx:30]: Set this as a const

File: src/main/webapp/resources/js/redux/endpoints/tags.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/**
+ * @fileoverview All `Tags` used in redux-toolkit should be listed here and export in the `PROVIDED_TAGS` array.
+ * This is consumed by the top level API.
+ */
+
+export const TAG_ANNOUNCEMENT_COUNT = "tag-announcement-count";
+export const TAG_COUNT = "tag-cart-count";
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/redux/endpoints/tags.ts:7]: Change to TAG_CART_COUNT

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
This is looking great! Just a few small comments :)

File: src/main/webapp/resources/js/redux/endpoints/tags.ts (hunk 2 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/redux/endpoints/tags.ts]: Updated in 5b877f5  good catch :\_

File: src/main/webapp/resources/js/components/main-navigation/components/CurrentUser.tsx (hunk 1 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/components/main-navigation/components/CurrentUser.tsx]: Updated in 14d55df

File: src/main/webapp/resources/js/redux/endpoints/tags.ts (hunk 2 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/redux/endpoints/tags.ts]: :)

**Review by deepsidhu85 — APPROVED** [review]

**Pull request merged**